webclient/src/components/Toast/ToastContext.tsx

```diff
@@ -0,0 +1,71 @@
+import { createContext, FC, ReactChild, ReactNode, useContext, useEffect, useReducer } from 'react'
+
+import { ACTIONS, initialState, reducer } from './reducer';
+import Toast from './Toast'
+
+interface Toast {
+  isOpen: boolean,
+  children: ReactChild,
+}
+
+interface ToastContext {
+    toasts: Toast[],
+    addToast: (key, children) => void,
+    openToast: (key) => void,
+    closeToast: (key) => void,
+    removeToast: (key) => void,
+}
+
+const ToastContext: any = createContext({
+  toasts: new Map(),
+  addToast: (key, children) => {},
+  openToast: (key) => {},
+  closeToast: (key) => {},
+  removeToast: (key) => {},
+});
+
+export const ToastProvider: FC<ReactNode> = (props) => {
+  const { children } = props
+  const [state, dispatch] = useReducer(reducer, initialState)
+  const providerState = {
```

Comment:
```suggestion
  const providerState: ToastContext = {
```

Reply:
Also, why are these methods defined here in a separate object and not defined in the params passed to `createContext`? Those seem like implementations, but they're noops.

Reply:
`createContext` accepts a `defaultValue` which needs to match the "shape" that the consumers expect.  [The docs](https://reactjs.org/docs/context.html#updating-context-from-a-nested-component) cover this briefly.

In my own words, the default value needs to have all of the same properties and methods that the final state will have or we risk runtime errors.